### ISSUE TITLE
Auth: non-serializable value detected in an login action using redux-starter-kit

### ISSUE THREAD

**Issue by fuhrmannb** [prescottprue/react-redux-firebase#761]
**Do you want to request a *feature* or report a *bug*?**  

Bug

**What is the current behavior?**

Using `redux-starter-kit` in my program, this library alerts me after a successful login that an non-serializable value is detected in the action `@@reactReduxFirebase/LOGIN`:

![Screenshot_20190910_160832](https://user-images.githubusercontent.com/19579566/64622253-4f214f80-d3e7-11e9-9b90-a9ae5571d3e8.png)

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [codesandbox](https://codesandbox.io/) or similar.**

Setup an simple auth application using `redux-starter-kit` with the store configured as follow:

```typescript
import { configureStore } from "redux-starter-kit"

import firebase from "firebase/app"
import "firebase/auth"
import "firebase/firestore"
import { createFirestoreInstance } from "redux-firestore"

firebase.initializeApp(myFirebaseConfig)
firebase.firestore()

const store = configureStore({
  reducer: myAppReducer,
})

export const rrfProps = {
  firebase,
  config: {
    userProfile: 'users',
    useFirestoreForProfile: true,
  },
  dispatch: store.dispatch,
  createFirestoreInstance
}
```

```tsx
<Provider store={appStore}>
  <ReactReduxFirebaseProvider {...rrfProps}>
    <MyApp />
  </ReactReduxFirebaseProvider>
</Provider>
```
**What is the expected behavior?**

Should the library use only serializable object into Redux action? That is expected by [Redux FAQ](https://redux.js.org/faq/actions#why-should-type-be-a-string-or-at-least-serializable-why-should-my-action-types-be-constants). Or should I make an exception for this specific case?

**Which versions of dependencies, and which browser and OS are affected by this issue? Did this work in previous versions or setups?**

OS: Manjaro Linux (last stable version)
Browser: Firefox 69.0 (64-bit)
Package versions:
* "react": "^16.8.6"
* "react-firebaseui": "^4.0.0",
* "react-redux": "^7.1.0"
* "react-redux-firebase": "^3.0.0-beta.2"
* "react-router-dom": "^5.0.1"
* "redux": "^4.0.4"
* "redux-firestore": "^0.8.0"
* "redux-starter-kit": "^0.6.3"

**Comment by prescottprue** [comment]
Interesting thanks for reporting, I'll look into if there is a fix that won't break things.

As noted [in this comment](https://github.com/rt2zz/redux-persist/issues/988#issuecomment-529407939) you should be able to configure the middleare to remove the serilizable check or ignore specific actions:

**Disable Check**
```js
import { configureStore, getDefaultMiddleware } from 'redux-starter-kit'

export const store = configureStore({
  reducer,
  middleware: getDefaultMiddleware({
    serializableCheck: false,
  }),
})
```

**Disable Certain Actions (Such as `LOGIN`)**
```js
import { actionTypes } from 'react-redux-firebase'
export const store = configureStore({
  reducer,
  middleware: getDefaultMiddleware({
    serializableCheck: {
        ignoredActions: [actionTypes.LOGIN]
    }
  }),
})
```

**Comment by fuhrmannb** [comment]
Thanks for this workaround.
I just want to notice that this workaround only work with `redux-starter-kit` v0.7.0, the last current version released 3 days ago (which add the ability to set options to `getDefaultMiddleware`)

**Comment by fuhrmannb** [comment]
I have also this error to the following actions:
* `FILE_UPLOAD_START`
* `FILE_UPLOAD_ERROR`
* `FILE_UPLOAD_COMPLETE`
* `FILE_DELETE_START`
(and I think to all related `FILE_*` event from storage)

**Comment by sebitoelcheater** [comment]
Rather than disabling `serializableCheck`, shouldn't the serialization problem be solved?

**Comment by prescottprue** [comment]
@sebitoelcheater Yes - open to PRs if you get a chance, but I have been too busy recently to look back into this

**Comment by kanlanc** [comment]
So, should the dev tool kit be removed for the package to work properly?
